src/modules/ekf2/EKF/aid_sources/gnss/gps_checks.cpp

```diff
@@ -62,19 +58,17 @@ void Ekf::collect_gps(const gnssSample &gps)
 {
 	if (_filter_initialised && !_NED_origin_initialised && _gps_checks_passed) {
 		// If we have good GPS data set the origin's WGS-84 position to the last gps fix
-		const double lat = gps.lat;
-		const double lon = gps.lon;
-
-		if (!_pos_ref.isInitialized()) {
-			_pos_ref.initReference(lat, lon, gps.time_us);
-
-			// if we are already doing aiding, correct for the change in position since the EKF started navigating
-			if (isHorizontalAidingActive()) {
-				double est_lat;
-				double est_lon;
-				_pos_ref.reproject(-_state.pos(0), -_state.pos(1), est_lat, est_lon);
-				_pos_ref.initReference(est_lat, est_lon, gps.time_us);
-			}
+		//  force new origin if it was previously set early by rough 2d fix (_gpos_origin_eph > 0)
+		if (!_pos_ref.isInitialized() || (_gpos_origin_eph > 0.f)) {
```

Comment:
When would it then not enter that `if` case?

Reply:
When the pos_ref is set from a command the eph is 0, when set earlier with the crude 2d fix it's > 0.

Maybe it would be better to track the different cases explicitly.
Origin set automatically from GPS, manually from command, etc